### PR TITLE
Support editing/patching model files

### PR DESCRIPTION
**Pull request by nkoenig**
This allows an `edit` command to change a model's files, metadata, tags, etc.

Example usage:

```
ign fuel edit -v 4 -m PATH_TO_MODEL -u "https://fuel.ignitionrobotics.org/1.0/OWNER/models/MODEL_NAME" --header 'Private-Token: PRIVATE_TOKEN'
```

Signed-off-by: Nate Koenig <nate@openrobotics.org>

### CODE REVIEW

File: include/ignition/fuel_tools/FuelClient.hh [hunk 1]
```diff
@@ -348,6 +348,21 @@ namespace ignition
                   const ignition::fuel_tools::ModelIdentifier &_model,
                   const std::vector<std::string> &_headers);
 
+      /// \brief Update a model using a PATCH request.
+      ///
+      /// Model fields that are patched by this function:
+      ///   * private
```
Comment (chapulina) [inline include/ignition/fuel_tools/FuelClient.hh:354]: Is this correct? The files are also patched, right?

File: src/ign.hh [hunk 2]
```diff
@@ -121,6 +121,7 @@ extern "C" IGNITION_FUEL_TOOLS_VISIBLE int pbtxt2Config(const char *_path);
 /// \return 1 if successful, 0 if not.
 extern "C" IGNITION_FUEL_TOOLS_VISIBLE int editUrl(
     const char *_url, const char *_header = nullptr,
-    const char *_private = nullptr);
+    const char *_private = nullptr,
+    const char *_path = nullptr);
```
Comment (chapulina) [inline src/ign.hh:125]: Missing doxygen

**Review by chapulina — COMMENTED** [review]
I understand that testing a successful edit would be complicated, but maybe we could at least add some tests for some of the failure modes, to cover part of the code.

File: include/ignition/fuel_tools/FuelClient.hh (hunk 1 above)
Comment (nkoenig) [reply include/ignition/fuel_tools/FuelClient.hh:354]: Update in e3ae7d852192597e06022dbb5a3e9a03cdca5485

File: src/ign.hh (hunk 2 above)
Comment (nkoenig) [reply src/ign.hh:126]: Added in e3ae7d852192597e06022dbb5a3e9a03cdca5485

**Comment by nkoenig** [comment]
> I understand that testing a successful edit would be complicated, but maybe we could at least add some tests for some of the failure modes, to cover part of the code.

Some tests added in e3ae7d852192597e06022dbb5a3e9a03cdca5485

**Review by chapulina — APPROVED** [review]
Works for me :+1: 

One weird thing I ran into was that I forgot to pass `-u` to the edit command, and I got this error message:

`Invalid URL: only models can be edited right now.`

I think we could catch the missing URL earlier.

**Comment by nkoenig** [comment]
> Works for me 
> 
> One weird thing I ran into was that I forgot to pass `-u` to the edit command, and I got this error message:
> 
> `Invalid URL: only models can be edited right now.`
> 
> I think we could catch the missing URL earlier.

Added early catch in aa8250c2ed05db66aa541aa9dc68afd90d6de78a.

**Pull request merged**